bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/auth/client/oauth2/CipherTest.java

```diff
@@ -0,0 +1,57 @@
+package org.eclipse.smarthome.core.auth.client.oauth2;
+
+import static org.junit.Assert.assertEquals;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+
+import org.junit.Test;
+
+public class CipherTest {
+
+    private static final String ENCRYPTION_ALGO = "AES/CBC/PKCS5Padding";
+
+    @Test
+    public void testEncDec() throws IllegalBlockSizeException, BadPaddingException, InvalidKeyException,
+            NoSuchAlgorithmException, NoSuchPaddingException, InvalidAlgorithmParameterException {
+        Cipher cipherEnc = Cipher.getInstance(ENCRYPTION_ALGO);
+        Cipher cipherDec = Cipher.getInstance(ENCRYPTION_ALGO);
+
+        byte[] iv = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
+        IvParameterSpec ivspec = new IvParameterSpec(iv);
+
+        SecretKey encryptionKey = generateEncryptionKey();
+
+        cipherEnc.init(Cipher.ENCRYPT_MODE, encryptionKey, ivspec);
+        cipherDec.init(Cipher.DECRYPT_MODE, encryptionKey, ivspec);
+
+        byte[] encrypted = cipherEnc.doFinal("hello world".getBytes());
+        byte[] decrypted = cipherDec.doFinal(encrypted);
+
+        assertEquals("Decrypted string should be same as before", "hello world", new String(decrypted));
+    }
+
+    private static SecretKey generateEncryptionKey() throws NoSuchAlgorithmException {
+        KeyGenerator keygen = KeyGenerator.getInstance("AES");
+        keygen.init(128); // cannot use very high grade encryption due to jvm limit
+        SecretKey secretKey = keygen.generateKey();
+        return secretKey;
+    }
+
+    @Test
+    public void testIV() {
+        byte[] ivspec = new byte[] { 12, 43, 98, -55, 127, -88, 33, 90, -111, -6, -15, 74, -32, -24, 69, 24 };
+
+        System.out.println("IV=" + Base64.getEncoder().encodeToString(ivspec));
```

Comment:
A test case without assertion

Reply:
the idea was that it should not throw exception.